R/release.R

```diff
@@ -366,24 +366,24 @@ get_release_news <- function(SHA = gert::git_info(repo = git_repo())$commit,
 
   if (HEAD == SHA) {
     news_path <- proj_path("NEWS.md")
-  } else {
-    news_path <- withr::local_tempfile()
-    gh <- purrr::possibly(gh_tr(tr), otherwise = NULL)
-    gh(
-      "/repos/{owner}/{repo}/contents/{path}",
-      path = "NEWS.md", ref = SHA,
-      .destfile = news_path,
-      .accept = "application/vnd.github.v3.raw"
-    )
+    if (file_exists(news_path)) {
+      return(news_latest(read_utf8(news_path)))
+    } else {
+      return("Initial release")
```

Comment:
Or just error? I don't think we can do anything useful without news bullets?

Reply:
In my scenario, I hadn't realised that the initial release of my package to CRAN was supposed to include a NEWS file (it's not obvious that "first release" counts as news, I guess). I wanted to create a github release but got an error about missing news. I tried creating NEWS.md retrospectively but the error persisted until I realised that it was using the repository state from the CRAN release commit.

That leaves me kind of stuck. There's no way to recover because I can't change history and alter the released commit. The only option is to create the release manually on github and double check that the commit matches. That's no big deal, but it's slightly jarring because `usethis` already knows exactly which commit I want to release but is refusing to do it because of something I can't change. I think offering to create an empty draft release would be a slightly friendlier fallback than just erroring.

Reply:
I think it would be better to handle that problem by prompting earlier in the release checklist to call `use_news_md()`.

Reply:
Fair enough. In `devtools::release()` it asks "Have you updated `NEWS.md` file?", but I just said yes because in my mind there was no news since it was an initial release. It didn't complain that the news file was missing, and I didn't realise that it would have consequences later.

Reply:
> That leaves me kind of stuck. There's no way to recover ... but is refusing to do it because of something I can't change.

This is a phenomenon we've grappled with elsewhere in usethis (mostly because of Git/GitHub stuff) and, as a rule, I try to eliminate these deadends. This PR should now soldier on with release, but provide lots of signals to create `NEWS.md` for future happiness.

#1691 and #1687 show that many folks seems to bang their shins on this specifically.